ksqldb-functional-tests/src/test/resources/query-validation-tests/decimal.json

```diff
@@ -64,7 +64,7 @@
         {"topic": "OUTPUT", "value": {"DEC": 0.1}},
         {"topic": "OUTPUT", "value": {"DEC": 0.01}},
         {"topic": "OUTPUT", "value": {"DEC": 0.001}},
-        {"topic": "OUTPUT", "value": {"DEC": 0.0001}}
+        {"topic": "OUTPUT", "value": {"DEC": 0.00010}}
```

Comment:
shouldn't we enable the test below? also why does this change `0.00010` is not `DECIMAL(6,4)`. Seems like a bug?

Reply:
Discussed offline. This is happening for a few reasons, none of which is a regression that this patch introduces.

First, we fixed the serialization bug where we chop off trailing zeroes, so now trailing zeroes that are in the decimal values being serialized are showing up. But why are there trailing zeroes?

Our test code takes the value from the input topic and deserializes/serializes it from/to json a couple times before it puts it in the source topic. In one of those serializations, it converts it to a double value in the form "10E-5". ksqlDB's json deserializer then picks it up and converts it to the decimal "0.00010", and eventually writes that decimal to the sink. So one reason is that the test code is changing the value in the source.

Still, I would think our deserializer would coerce the received value into the decimal scale defined in the schema, but it's not doing that. I think this might be a bug (we can discuss in a follow-up).

I'm going to merge this PR because I don't think it's adding new regressions, and then file follow-ups to address the above 2 issues.